addons/JobChange/jobchange.lua

```diff
@@ -27,119 +27,181 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
-                    jobchange(jobid,main_sub)
-                end
-            end
-        else
-            windower.add_to_chat(4,"JobChange: Not close enough to a Moogle!")
-        end        
-    else
-        windower.add_to_chat(4,"JobChange: Could not change "..command.." to "..job:upper().." ---Mistype|NotUnlocked")
-    end
-    
-end)
+local jobchange = function(job, main)
+    local packet = packets.new('outgoing', 0x100, {
+        [(main and 'Main' or 'Sub')..' Job'] = job,
+    })
+    packets.inject(packet)
+end
 
-function find_conflict(job,self)
-    if self.main_job == job:upper() then
-        return "main"
+local find_conflict = function(job_name, p)
+    if p.main_job == job_name:upper() then
+        return 'main'
     end
-    if self.sub_job == job:upper() then
-        return "sub"
+    if p.sub_job == job_name:upper() then
+        return 'sub'
     end
 end
 
-function find_temp_job(self)
-    local starting_jobs = S { 'WAR', 'MNK', 'WHM', 'BLM', 'RDM', 'THF' } 
-    for i in pairs(starting_jobs) do
-        if not find_conflict(i,self) then 
+local find_temp_job = function(p)
+    for _, job_name in ipairs(temp_jobs) do
+        if not find_conflict(job_name, p) and p.jobs[job_name:upper()] > 0 then 
             for index,value in pairs(res.jobs) do
-                if value.ens == i then
+                if value.ens == job_name then
                     return index
                 end
             end
         end
     end
 end
 
-function find_job(job,self)
-    local jobLevel = self.jobs[job:upper()]
+local find_job = function(job,p)
+    if job == nil then return nil end
+    local jobLevel = p.jobs[job:upper()]
     for index,value in pairs(res.jobs) do
         if value.ens:lower() == job and jobLevel > 0 then 
             return index
         end
     end
 end
 
-
-function find_job_change_npc()
+local find_job_change_npc = function()
     local info = windower.ffxi.get_info()
-    if not (info.mog_house or S{'Selbina', 'Mhaura', 'Tavnazian Safehold', 'Nashmau', 'Rabao', 'Kazham', 'Norg'}:contains(res.zones[info.zone].english)) then
-        windower.add_to_chat(4,'JobChange: Not in a zone with a Change NPC')
+    if not (info.mog_house or mog_zones:contains(res.zones[info.zone].english)) then
+        log('Not in a zone with a Change NPC')
         return
     end
 
-    for i, v in pairs(windower.ffxi.get_mob_array()) do
-        if v.distance < 36 and S{'Moogle', 'Nomad Moogle', 'Green Thumb Moogle'}:contains(v.name) then
+    for _, v in pairs(windower.ffxi.get_mob_array()) do
+        if v.distance < 36 and v.valid_target and moogles:contains(v.name) then
             return v
         end
     end
 end
+
+windower.register_event('addon command', function(command, ...)
+    local p = windower.ffxi.get_player()
+    local args = L{...}
+    local job = ''
+    if args[1] then 
+        job = args[1]:lower()
+    end
+    local main = nil
+    local sub = nil
+    if command:lower() == 'main' then
```

Comment:
`command` should be lowercased at the start, so it does not need to be done in every check. I would also uppercase the job strings, so it does not need to be performed everywhere below. Would make the code more readable imo, but up to you.